tests/ipfwd/test_nhop_count.py

```diff
@@ -0,0 +1,178 @@
+    """
+    Create a list of next hop paths with given IP, MAC parameters
+    """
+    def __init__(
+        self, duthost, asic, count, iface, ip="172.16", mac="C0:FF:EE:00"
+    ):
+        IP_MAC = namedtuple('IP_MAC', 'ip mac')
+        self.iface = iface
+        self.arp_list = []
+        self.duthost = duthost
+        self.asic = asic
+        self.if_addr = "{}.0.3/16".format(ip)
+        self.ip = None
+        self.fileloc = os.path.join(os.path.sep, "tmp")
+        self.filename = os.path.join(self.fileloc, "static_arp.sh")
+
+        for i in range(11, count+11):
+            moff1 = "{0:x}".format(i/255)
+            moff2 = "{0:x}".format(i%255)
+
+            ipoff1 = i / 255
+            ipoff2 = i % 255
+            self.arp_list.append(IP_MAC(
+                "{}.{}.{}".format(ip, ipoff1, ipoff2),
+                "{}:{}:{}".format(mac, moff1, moff2)
+            ))
+
+        ip_iface = "ip address add {} dev {}".format(self.if_addr, self.iface)
+
+        logging.info("IF ADDR ADD {}".format(ip_iface))
+        # add IP address to the eth interface
+        result = asic.command(ip_iface)
+        pytest_assert(result["rc"] == 0, ip_iface)
+
+    def arps_add(self):
+        """
+        Create a file with static arp add commands, copy file
+        to DUT and run it from DUT
+        """
+        arp_cmd = "sudo {} arp -s {} {}"
+        with open(self.filename, "w") as fn:
+            for ip_mac in self.arp_list:
+                cmd = arp_cmd.format(self.asic.ns_arg, ip_mac.ip, ip_mac.mac)
+                fn.write(cmd + "\n")
+
+        self.duthost.copy(src=self.filename, dest=self.filename, mode=0755)
+        result = self.duthost.shell(self.filename)
+        pytest_assert(
+            result["rc"] == 0,
+            "arp add failed on duthost:{}".format(self.filename)
+        )
+        self.duthost.shell("rm {}".format(self.filename))
+
+    def arps_del(self):
+        """
+        Create a file with static arp del commands, copy file
+        to DUT and run it from DUT
+        """
+        arp_cmd = "sudo {} arp -d {}"
+        with open(self.filename, "w") as fn:
+            for ip_mac in self.arp_list:
+                cmd = arp_cmd.format(self.asic.ns_arg, ip_mac.ip)
+                fn.write(cmd + "\n")
+
+        self.duthost.copy(src=self.filename, dest=self.filename, mode=0755)
+        try:
+            self.duthost.shell(self.filename)
+            self.duthost.shell("rm {}".format(self.filename))
+        except:
+            pass
+
+        os.remove(self.filename)
+
+    def add_ip_route(self, ip="192.168.5.0/24"):
+        """
+        Add IP route with ECMP paths via 'vtysh' command
+        """
+        self.arps_add()
+        self.ip = ip
+        vty_cmd = "-c 'config term'"
+
+        for ip_mac in self.arp_list:
+            vty_cmd += " -c 'ip route {} {}'".format(self.ip, ip_mac.ip)
+
+        vty_cmd += " -c 'exit'"
+
+        result = self.asic.run_vtysh(vty_cmd)
```

Comment:
instead of invoking should we use SONiC static route programming. This will test bgpcfgd also.

Reply:
Not sure static route is supported in all releases. For this test scenario this should be fine. Do you know if static route is supported in all releases.